### ISSUE TITLE
Syncing slices of redux state with subquery results

### ISSUE THREAD

**Issue by talolard** [rtk-incubator/rtk-query#84]
Relates to #79 

I [commented on twitter](https://twitter.com/de_stroy/status/1334191220927590400) about how integrating RQ with redux hurts. 
I'll describe what's hurting for us and add a video below, I have problems but no good solutions to offer: 

We have a data model and API calls that look like this
```ts
export interface X {
    id:string
    data:Record<any, any>
}
interface XSummary {
    xId:string
    stat1:number,
    stat2:number
    
}
export interface RelatesToX {
    id:string
    xId:string //The id of the X
}
function getAllXSummary():Array<XSummary>
/* Expensive to call because it goes over all the X */ 

function getOneXSummary(xId:string) :XSummary

function getXWithRelated() :{x:X,related:Array<RelatesToX>} 

function changeARelatedToX(relatedToXId:string,something:any)
```

We display the list of XSummary to the user and when they click on one we should them the X with all of the RelatedToX.

When the user calls ***changeARelatedToX*** we'd like to update the single ***XSummary*** in the list, without calling **getAllXSummary** because it's expensive. 

Ideally, when the user triggers a call to ***changeARelatedToX*** it would subsequently call ***getOneXSummary*** for the given xId. 

# Why It's Hard with React-Query

React Query caches the the list of XSummary somewhere magical, so it's not clear what the result of ***getOneXSummary*** should update. Once the call to ***getAllXSummary*** is complete, we want to keep track of the individual XSummary states so that we can make the user happy (video below), but it's just not clear how to do that

# Real World User Case
I made a loom of where we are struggling. 
https://www.loom.com/share/91ce26804c8740e1bc1f741f067a9b9b

# Towards a solution

This seems like a variation on #79. I want my list query to register it's results in a clearly indicated state slice, and be able to associate a query with each item in that slice, where the originating query to populate the slice is the query that brought back the entire list. 

Then, I'd also like a canonical way to trigger the "getter" for a particular item in the slice programmatically from some seemingly unrelated query. 

# Why that's hard in RQ
Possibly repeating myself, but in react-query I don't have a clear way to 
1. associate an item that came back from a "list query" with it's own query. e.g. what is the query key for that item ? 
2. Associate each item in a (displayed) list with it's own query that is pre-propulated with the results of the original list query.

**Comment by phryneas** [comment]
So, from the back of my head, right now with the existing APIs, you would have two options:

1. modify the list in "the cache"
or
2. keep a slice with data from the list and modify that

Option 1 would work well as long as your `getAllXSummary` returns really all X, meaning without paginagion & other changing arguments
```js
dispatch(
  api.util.updateQueryResult(
    'getAllXSummary', 
    undefined, // the argument you passed to useGetAllXSummaryQuery - undefined if none
    (draft) => {
      // just modify draft here, it's the "cache" for your `getAllXSummary` query call with argument `undefined`
    })
)
```

Option 2 would mean you have your own slice and add data once the query finishes
```js
const slice = createSlice({
//...
  extraReducers(builder) {
    builder.addMatcher(api.endpoints.getAllXSummary.onSuccess, (state, action) => {
      // handle your slice yourself
    }
  }
})
```
this is pretty much how you would also do it with a `createAsyncThunk` and now you are responsible for that data yourself - including potential cleanup.

**Comment by phryneas** [comment]
I'm closing everything here now. If there's any need for further discussion, that should take place in a new issue over at https://github.com/reduxjs/redux-toolkit/
